vendor_blueprints/refresh_datasource.py

```diff
@@ -0,0 +1,87 @@
+import tableauserverclient as TSC
+import argparse
+import sys
+
+EXIT_CODE_FINAL_STATUS_SUCCESS = 0
+EXIT_CODE_UNKNOWN_ERROR = 3
+EXIT_CODE_INVALID_CREDENTIALS = 200
+EXIT_CODE_INVALID_RESOURCE = 201
+EXIT_CODE_FINAL_STATUS_ERRORED = 211
+EXIT_CODE_FINAL_STATUS_CANCELLED = 212
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', dest='username', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--site_id', dest='site_id', required=True)
+    parser.add_argument('--content_url', dest='content_url', required=True)
+    parser.add_argument('--datasource_id', dest='datasource_id', required=True)
+    args = parser.parse_args()
+    return args
+
+def authenticate_tableu(username, password, site_id, content_url):
+    """TSC library to sign in and sign out of Tableau Server and Tableau Online.
+
+    :param username:The name of the user.
+    :param password:The password of the user.
+    :param site_id: The site_id for required datasources. ex: ffc7f88a-85a7-48d5-ac03-09ef0a677280
+    :param content_url: This corresponds to the contentUrl attribute in the Tableau REST API.
+    :return: connection object
```

Comment:
Being picky, but the spacing isn't consistent after the `:param xxx: ` declarations. 

Also, in general, all Python code should be pep8 formatted with `autopep8`. So I'd suggest adding something to your IDE to auto format on save or in a git pre-commit hook.

Reply:
checked with pylint